webapp/src/Utils/Utils.php

```diff
@@ -1022,6 +1022,32 @@ public static function generatePassword() : string
         return $rand_str;
     }
 
+    /**
+     * Generate a strong random password of the given length
+     *
+     * @param int $length
+     *
+     * @return string
+     */
+    public static function generateStrongPassword($length = 16) : string
+    {
+        $chars = str_split(
+            'abcdefghijklmnopqrstuvwxyz' .
```

Comment:
what about something like this?

```
array_merge(
  range('a', 'z'),
  range('A', 'Z'),
  range('0', '9')
)
```

Reply:
I like the range. I can make the length non-configurable indeed

Reply:
https://www.ncsc.nl/onderwerpen/authenticatie
I think you should atleast extend with the symbols in asci (if your terminal would not accept unicode symbols) and the length should be variable between 10 and 16 so the amount of possible passwords is still high and bruteforce really becomes undoable?

I think even better if you want to keep the passwords safe is: https://www.zdnet.com/article/fbi-recommends-passphrases-over-password-complexity/ basicly the comic advice we all know so people might actually remember the password so its not downgraded to "test" as this might be remembered (although this really will create different discussions)

Reply:
Regarding password strength, with 64 options for each character and a length of 16 the number of possible passwords is 64^16 = 7.9 * 10^28. The paper [SHA-1 is a Shambles](https://sha-mbles.github.io/) mentions a cost of 100k USD to compute 2^64 operations of symmetric cryptography, so the cost of cracking such a password is on the order of 10^14 USD and that's plenty.

Using a variable shorter length might seem like it would significantly increase the number of possible passwords, but surprisingly it doesn't: sum(64^n for n in 10..16) = 8.05 * 10^28. It's pretty hard to make the distribution of passwords uniform when the length is variable.

Passphrases are cool but harder to implement because they need a word list, and they are still difficult to memorize.

I think the current scheme (which is already used for other passwords) is fine. Users can easily copy-paste the password or change it.

Blacklisting weak passwords is a good idea. It might make people angry when creating lots of dummy accounts for local testing :)

Reply:
I indeed took the same scheme as used in genadminpasswords, which is basically the characters from base64 + replace `/` with `_` and `+` with `-`.

If we want to change the password generation let's do that later in a separate PR.